cmds/core/sluinit/uinit.go

```diff
@@ -48,19 +49,20 @@ func checkDebugFlag() {
 func main() {
 	checkDebugFlag()
 
+	defer unmountAndExit() // called only on error, on success we kexec
 	slaunch.Debug("********Step 1: init completed. starting main ********")
 	tpmDev, err := tpm.GetHandle()
 	if err != nil {
 		log.Printf("tpm.getHandle failed. err=%v", err)
```

Comment:
this should be log.Fatal in this and all succeeding similar cases.

Reply:
@rminnich wait...I am confused...I thought log.Fatal doesn't call deferred functions. That would mean unmountAndExit would never get callled on error exit ? we want the devices to be unmounted on error exit...

Its discussed on this thread: https://stackoverflow.com/questions/17888610/are-deferred-functions-called-when-calling-log-fatalln
and here: https://quasilyte.dev/blog/post/log-fatal-vs-log-panic/

Reply:
@rminnich wait...I am confused...I thought log.Fatal doesn't call deferred functions. That would mean unmountAndExit would never get callled on error exit ? we want the devices to be unmounted on error exit...

Its discussed on this thread: https://stackoverflow.com/questions/17888610/are-deferred-functions-called-when-calling-log-fatalln
and here: https://quasilyte.dev/blog/post/log-fatal-vs-log-panic/

Reply:
yeah you were previously calling os.Exit(1) which is why I recommended log.Fatal. If you want to do it this way that's fine.